prow/plugins/project/project.go

```diff
@@ -0,0 +1,257 @@
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	projectMaintainers, err := gc.ListTeamMembers(projectConfigMap[repo].ProjectMaintainersID, github.RoleAll)
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, person := range projectMaintainers {
+		login := github.NormLogin(e.User.Login)
+		if github.NormLogin(person.Login) == login {
+			found = true
+			break
+		}
+	}
+	if !found {
+		// not in the project maintainers team
+		msg := fmt.Sprintf(notATeamMemberMsg, org, repo, org, repo)
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	repos, err := gc.GetRepos(org, false)
+	if err != nil {
+		log.WithError(err).Errorf("Error listing all repos in org %s", org)
+		return err
+	}
+	// Get all projects for all repos
+	var projects []github.Project
+	for _, repo := range repos {
+		repoProjects, err := gc.GetRepoProjects(org, repo.Name)
+		if err != nil {
+			log.WithError(err).Errorf("Error listing all projects under repo %s", repo.Name)
+			return err
+		}
+		projects = append(projects, repoProjects...)
+	}
+
+	// Only fetch org projects if we can't find the proposed project / project to clear in the repo projects
+	projectToFind := proposedProject
+	if proposedProject == clearKeyword {
+		projectToFind = projectToClear
+	}
+	updateProjectNameToIDMap(projects)
+	var projectID int
+	var ok bool
+	if projectID, ok = projectNameToIDMap[projectToFind]; !ok {
+		// Get all projects for this org
+		orgProjects, err := gc.GetOrgProjects(org)
+		if err != nil {
+			log.WithError(err).Errorf("Error getting projects for org %s", org)
+			return err
+		}
+		projects = append(projects, orgProjects...)
+
+		// If still can't find proposed project / project to clear in the list of projects, abort and create a comment
+		updateProjectNameToIDMap(projects)
+		if projectID, ok = projectNameToIDMap[projectToFind]; !ok {
+			slice := make([]string, 0, len(projectNameToIDMap))
+			for k := range projectNameToIDMap {
+				slice = append(slice, fmt.Sprintf("`%s`", k))
+			}
+			sort.Strings(slice)
+
+			msg := fmt.Sprintf(invalidProject, strings.Join(slice, ", "))
+			return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+		}
+	}
+
+	// Clear issue/PR from project if command is to clear
+	if proposedProject == clearKeyword {
+		if err := gc.DeleteProjectCard(projectID); err != nil {
+			log.WithError(err).Errorf("Error removing project card from project %s (ID %d) for %s/%s#%d.", projectToFind, projectID, org, repo, e.Number)
+			return err
+		}
+		return nil
+	}
+
+	// Get all columns for projectToFind
+	projectColumns, err := gc.GetProjectColumns(projectID)
+	if err != nil {
+		log.WithError(err).Errorf("Error listing all columns under project %s", projectToFind)
+		return err
+	}
+
+	// If proposedColumn is not found (or not provided), add to one of the default
```

Comment:
This should be documented in the configuration fields

Reply:
@cblecker is this surprising behavior? Should we just do the simple thing and error if we can't find the column that was requested?